src/main/java/edu/illinois/cs/dt/tools/fixer/CleanerFixerPlugin.java

```diff
@@ -186,13 +183,97 @@ private void applyFix(final String polluterName,
         //        }
     }
 
+    // The delta debugging logic to return minimal list of statements to add in to make victim pass
+    // In addition to polluter and victim to run, need also list of statements; assumed statements already make victim pass
+    // The value n is needed to represent granularity of partitioning the statements to get minimal list
+    private NodeList<Statement> deltaDebug(final String polluterName,
+                                           final JavaMethod victimMethod,
+                                           final NodeList<Statement> cleanerStmts,
+                                           int n) throws Exception {
+        // If n granularity is greater than number of cleaner statements, then finished, simply return cleaner statements
+        if (cleanerStmts.size() < n) {
+            return cleanerStmts;
+        }
+
+        // Cut the statements into n equal chunks and try each chunk
+        int chunkSize = (int)Math.round((double)(cleanerStmts.size()) / n);
+        List<NodeList<Statement>> chunks = new ArrayList<>();
+        for (int i = 0; i < cleanerStmts.size(); i += chunkSize) {
+            NodeList<Statement> chunk = NodeList.nodeList();
+            NodeList<Statement> otherChunk = NodeList.nodeList();
+            // Start complement chunk by grabbing statements before chunk start
+            for (int j = 0; j < i; j++) {
```

Comment:
Looks like these three loops here could be replaced by a call to `subList`, something like `otherChunk.addAll(cleanerStmts.subList(0, i))`, if I properly understand what's going on.